timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/v5/redirectionservice/DummyService.java

```diff
@@ -0,0 +1,105 @@
+package nl.knaw.huygens.timbuctoo.v5.redirectionservice;
+
+import nl.knaw.huygens.persistence.PersistenceException;
+import nl.knaw.huygens.persistence.PersistenceManager;
+import nl.knaw.huygens.timbuctoo.core.NotFoundException;
+import nl.knaw.huygens.timbuctoo.core.TransactionState;
+import nl.knaw.huygens.timbuctoo.core.dto.EntityLookup;
+import nl.knaw.huygens.timbuctoo.util.Tuple;
+import nl.knaw.huygens.timbuctoo.v5.dataset.AddTriplePatchRdfCreator;
+import nl.knaw.huygens.timbuctoo.v5.dataset.DataSetRepository;
+import nl.knaw.huygens.timbuctoo.v5.dataset.ImportManager;
+import nl.knaw.huygens.timbuctoo.v5.dataset.dto.DataSet;
+import nl.knaw.huygens.timbuctoo.v5.dataset.dto.DataSetMetaData;
+import nl.knaw.huygens.timbuctoo.v5.filestorage.exceptions.LogStorageFailedException;
+import nl.knaw.huygens.timbuctoo.v5.queue.QueueManager;
+import nl.knaw.huygens.timbuctoo.v5.redirectionservice.exceptions.RedirectionServiceException;
+import nl.knaw.huygens.timbuctoo.v5.util.RdfConstants;
+import org.slf4j.Logger;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+public class DummyService extends RedirectionService {
+  public static final String DUMMY_QUEUE = "dummy";
+  private static final Logger LOG = org.slf4j.LoggerFactory.getLogger(RedirectionService.class);
+  private static final String PERSISTENT_ID = RdfConstants.timPredicate("persistentUri");
+  private final DataSetRepository dataSetRepository;
+  private PersistenceManager manager;
+
+  public DummyService(QueueManager queueManager, DataSetRepository dataSetRepository) {
+    super(DUMMY_QUEUE, queueManager);
+    this.dataSetRepository = dataSetRepository;
+    LOG.info("Using dummy persistence manager instead of real server");
+    manager = new DummyPersistenceManager();
+  }
+
+  @Override
+  protected void oldSavePid(RedirectionServiceParameters params) throws PersistenceException, URISyntaxException {
+    URI uri = params.getUrlToRedirectTo();
+    LOG.info(String.format("Retrieving persistent url for '%s'", uri));
+    String persistentId = (manager.persistURL(uri.toString()));
+    URI persistentUrl = new URI(manager.getPersistentURL(persistentId));
+
+    transactionEnforcer.execute(timbuctooActions -> {
+        try {
+          timbuctooActions.addPid(persistentUrl, params.getEntityLookup());
+          LOG.info("committed pid");
+          return TransactionState.commit();
+        } catch (NotFoundException e) {
+          LOG.warn("Entity for entityLookup '{}' cannot be found", params.getEntityLookup());
+          try {
+            manager.deletePersistentId(persistentId);
+          } catch (PersistenceException e1) {
+            LOG.error("Cannot remove url with id '{}'", persistentId);
+          }
+          return TransactionState.rollback();
+        }
+      }
+    );
+  }
```

Comment:
Code can be simplified by not using the DummyPersistenceManager. Fake the creation of the persistentUrl in this class.